### PR TITLE
Refactor for push buckets and kind

### PR DESCRIPTION
**Pull request by thockin**
Multiple steps - please review commit-by-commit.

Refactor prod-storage script to make it easy to add a new "push" bucket in prod.

Fix output to be legible (if this is good we can repeat for other scripts)

Add kind as a push bucket.

The new output looks something like this (debug left in here):

```
$ ./ensure-prod-storage.sh 
Ensuring all prod projects
  Ensuring project exists: k8s-artifacts-prod
  billingAccountName: billingAccounts/018801-93540E-22A20E
  billingEnabled: true
  name: projects/k8s-artifacts-prod/billingInfo
  projectId: k8s-artifacts-prod
  Enabling the container registry API: k8s-artifacts-prod
  Enabling the container analysis API: k8s-artifacts-prod
  Ensuring the GCR repository: k8s-artifacts-prod
  Ensuring the GCR exists and is readable
    region us
    TIM ensure_gcr_repo k8s-artifacts-prod us
    region eu
    TIM ensure_gcr_repo k8s-artifacts-prod eu
    region asia
    TIM ensure_gcr_repo k8s-artifacts-prod asia
  Empowering GCR admins
    region us
    TIM empower_gcr_admins k8s-artifacts-prod us
    region eu
    TIM empower_gcr_admins k8s-artifacts-prod eu
    region asia
    TIM empower_gcr_admins k8s-artifacts-prod asia
  Empowering image promoter
    region us
    TIM empower_artifact_promoter k8s-artifacts-prod us
    region eu
    TIM empower_artifact_promoter k8s-artifacts-prod eu
    region asia
    TIM empower_artifact_promoter k8s-artifacts-prod asia
  Enabling the GCS API: k8s-artifacts-prod
  Ensuring the GCS bucket: gs://k8s-artifacts-prod
    Ensuring the GCS bucket exists and is readable
    TIM ensure_public_gcs_bucket k8s-artifacts-prod gs://k8s-artifacts-prod
    Ensuring the bucket retention policy is set
    TIM ensure_gcs_bucket_retention gs://k8s-artifacts-prod 10y
    Empowering GCS admins
    TIM empower_gcs_admins k8s-artifacts-prod gs://k8s-artifacts-prod
  Ensuring project exists: k8s-artifacts-prod-bak
  billingAccountName: billingAccounts/018801-93540E-22A20E
  billingEnabled: true
  name: projects/k8s-artifacts-prod-bak/billingInfo
  projectId: k8s-artifacts-prod-bak
```

### CODE REVIEW

File: infra/gcp/lib_util.sh [hunk 1]
```diff
@@ -38,3 +38,12 @@ function color() {
     echo "$@"
     _nocolor
 }
+
+# Indent each line of stdin.
+# example: <command> | indent
+function indent() {
+    IFS=''
+    while read X; do
+        echo "$X" | paste /dev/null - | expand -2
```
Comment (listx) [inline infra/gcp/lib_util.sh:47]: How about just `sed -e 's/^/    /'` ?

**Comment by thockin** [comment]
Funny, that was the first version, then I took a diversion to what I hoped
would be nicer, turned out wasn't and then did not revert.

Fixed now.

File: infra/gcp/ensure-prod-storage.sh [hunk 2]
```diff
@@ -88,52 +70,136 @@ ALL_PROD_PROJECTS=(
...
+# this because we want to grant minimal permissions, but there's no concept of
+# permissions on a "subdirectory" of a bucket.  If we had a GCS promoter akin
+# to the GCR promoter, we might have used a single bucket, but we don't have
+# that yet.
+#
+# $1: The GCP project to make the bucket
+# $2: The bucket, including gs:// prefix
+# $3: The group email to empower (optional)
+function ensure_prod_gcs_bucket() {
+        if [ $# -lt 2 -o $# -gt 3 -o -z "$1" -o -z "$2" ]; then
```
Comment (spiffxp) [inline infra/gcp/ensure-prod-storage.sh:129]: nit: over-indented

**Comment by thockin** [comment]
I am running this now, and will fix the nit.

**Comment by BenTheElder** [comment]
thanks!

**Comment by thockin** [comment]
Activated

**Pull request merged**